packages/esm-reports-app/src/components/run-report/run-report-form.component.tsx

```diff
@@ -124,8 +130,17 @@ const RunReportForm: React.FC<RunReportForm> = ({ closePanel }) => {
   }
 
   function handleOnDateChange(fieldName, dateValue) {
-    const date = new Date(dateValue).toLocaleDateString();
-    setReportParameters((state) => ({ ...state, [fieldName]: date }));
+    const formattedDate = formatLocalDateToString(dateValue);
+    setReportParameters((state) => ({ ...state, [fieldName]: formattedDate }));
+  }
+
+  function formatLocalDateToString(dateValue) {
+    const date = new Date(dateValue);
+    const year = date.getFullYear();
+    const month = String(date.getMonth() + 1).padStart(2, '0');
+    const day = String(date.getDate()).padStart(2, '0');
+
+    return `${year}-${month}-${day}`;
```

Comment:
We tend to prefer using dayjs, which makes this relatively easy, but for preference, we should be using the functions built into the framework such as [formatDate](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/docs/API.md#formatdate) to avoid hard-coding date formats and calendars where they are not appropriate.